miden/tests/integration/stdlib/collections/mmr.rs

```diff
@@ -0,0 +1,229 @@
+        // root, absolute_pos, leaf
+        (merkle_root1, 0, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 0u64))?),
+        (merkle_root1, 1, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 1u64))?),
+        (merkle_root1, 2, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 2u64))?),
+        (merkle_root1, 3, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 3u64))?),
+        (merkle_root1, 7, merkle_store.get_node(merkle_root1, NodeIndex::new(3u8, 7u64))?),
+        (merkle_root2, 8, merkle_store.get_node(merkle_root2, NodeIndex::new(1u8, 0u64))?),
+        (merkle_root2, 9, merkle_store.get_node(merkle_root2, NodeIndex::new(1u8, 1u64))?),
+    ];
+
+    for (root, absolute_pos, leaf) in examples {
+        let source = format!(
+            "use.std::collections::mmr
+
+            begin
+                push.{num_leaves} push.1000 mem_store # leaves count
+                adv_push.4 push.1001 mem_storew dropw # MMR first peak
+                adv_push.4 push.1002 mem_storew dropw # MMR second peak
+
+                push.1000 push.{pos} exec.mmr::get
+            end",
+            num_leaves = num_leaves,
+            pos = absolute_pos,
+        );
+
+        let test = build_debug_test!(source, &[], advice_stack, merkle_store.clone());
+
+        // the stack should be first the leaf element followed by the tree root
+        let stack: Vec<u64> = leaf
+            .iter()
+            .map(StarkField::as_int)
+            .rev()
+            .chain(root.iter().map(StarkField::as_int).rev())
+            .collect();
+        test.expect_stack(&stack);
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_mmr_tree_with_one_element() -> Result<(), MerkleError> {
+    // This test uses three merkle trees for the MMR, one with 8 elements, one with 2, and one with
+    // a single leaf. The test is ensure the single leaf case is supported, the other two are used
+    // for variaty
+    let leaves1 = &[1, 2, 3, 4, 5, 6, 7, 8];
+    let leaves2 = &[9, 10];
+    let leaves3 = &[11];
+
+    let mut merkle_store = MerkleStore::new();
+    let merkle_root1 = merkle_store.add_merkle_tree(init_merkle_leaves(leaves1))?;
+    let merkle_root2 = merkle_store.add_merkle_tree(init_merkle_leaves(leaves2))?;
+    let merkle_root3 = init_merkle_leaves(leaves3)[0];
+
+    // In the case of a single leaf, the leaf is itself also the root
+    let stack: Vec<u64> = merkle_root3
+        .iter()
+        .map(StarkField::as_int)
+        .rev()
+        .chain(merkle_root3.iter().map(StarkField::as_int).rev())
+        .collect();
+
+    let advice_stack: Vec<u64> = merkle_root3.iter().map(StarkField::as_int).collect();
+    let source = format!(
+        "use.std::collections::mmr
+
+        begin
+            push.{num_leaves} push.1000 mem_store # leaves count
+            adv_push.4 push.1001 mem_storew dropw # MMR first peak
+
+            push.1000 push.{pos} exec.mmr::get
+        end",
+        num_leaves = leaves3.len(),
```

Comment:
Is this correct here? isn't `num_leaves = leaves1.len() + leaves2.len() + leaves3.len()`?

Let's also add a test for when we have just a single leaf - i.e., the entire MMR is just `let leaves = &[1];`.

Reply:
this test has both a single leaf and the last leaf of a MMR with 8/2/1

this only has `leaves3` because it tests the case of a single leaf.

Reply:
Ah - makes sense! Could you add a short comment above this block of code to describe the nature of the test?

Reply:
added two small comments